src/object.js

```diff
@@ -1,170 +1,112 @@
-            return map.getCenter().lng;
-        }
-        this.goTo = function(lat, lon, zoom) {
-            map.setView([lat, lon], zoom);
+BlurredLocation = function BlurredLocation(options) {
+    options = options || {};
+    options.map = options.map || L.map('map');
+    options.addGrid = options.addGrid || require('./addGrid.js');
+
+    L.tileLayer("https://a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/{z}/{x}/{y}.png").addTo(options.map);
+    options.location = options.location || {
+      lat: 41.01,
+      lon: -85.66
+    };
+    options.zoom = options.zoom || 13;
+    options.map.setView([options.location.lat, options.location.lon], options.zoom);
+
+    this.getLat = function () {
+      return options.map.getCenter().lat;
+    }
+
+    this.getLon = function() {
+      return options.map.getCenter().lng;
+    }
+
+    this.goTo = function(lat, lon, zoom) {
+      options.map.setView([lat, lon], zoom);
+    }
+
+    this.geocode = function(string) {
+      var url = "https://maps.googleapis.com/maps/api/geocode/json?address="+string.split(" ").join("+");
+      var Blurred = $.ajax({
+          async: false,
+          url: url
+      });
+      // console.log(Object.getOwnPropertyNames(Blurred));
+      // console.log(Blurred.readyState);
+
+      var geometry = Blurred.responseJSON.results[0].geometry.location;
+      // console.log(geometry.lat);
+      options.map.setView([geometry.lat, geometry.lng],options.zoom);
+      return geometry;
+    }
+
+    this.getSize = function() {
+      return options.map.getSize();
+    }
+
+    this.addGrid = options.addGrid;
+}
+
+function geoLocateFromInput(selector) {
+  var input = document.getElementById(selector);
+
+  var autocomplete = new google.maps.places.Autocomplete(input);
+  autocomplete.addListener('place_changed', function() {
+    setTimeout(function () {
+      var str = input.value;
+      var loc = str.split(' ').join('+');
+      $.getJSON("https://maps.googleapis.com/maps/api/geocode/json?address=" + loc + "&key=AIzaSyDWgc7p4WWFsO3y0MTe50vF4l4NUPcPuwE", function(data){
+        if(data.results[0]) {
+            var lat = data.results[0].geometry.location.lat;
+            var lng = data.results[0].geometry.location.lng;
+            panMap(lat, lng);
         }
-        this.geocode = function(string) {
-            $.ajaxSetup({ async: false });
-            var url = "https://maps.googleapis.com/maps/api/geocode/json?address=" + string.split(" ").join("+");
-            var temp = $.ajax({
-                async: false,
-                url: url
-            });
-            var geometry = temp.responseJSON.results[0].geometry.location;
-            var default_zoom = 13;
-            console.log(geometry.lat);
-            map.setView([geometry.lat, geometry.lng], default_zoom);
-            return geometry;
+      });
+    }, 10);
+  });
+};
+
+  function geoLocateFromLatLng(lat,lng) {
+    var lat = document.getElementById(lat);
```

Comment:
Also - once this big PR is merged, we can hopefully limit ourselves to each PR dealing with one isolated method, and its matching test. That way each PR can be reviewed quickly and easily by comparing the single method to the test that describes its intended use and output. Make sense?